### PR TITLE
Fix for deleted forms

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan @esoergel i get a key error on two form ids when i run the command in dry run. i'm assuming that's cause they were deleted or something. don't think it's too bad if we skip them

### CODE REVIEW

File: corehq/apps/cleanup/management/commands/fix_forms_with_missing_xmlns.py [hunk 1]
```diff
@@ -134,13 +134,16 @@ def fix_apps(unique_id_to_xmlns_map, app_to_unique_ids_map, log_file, dry_run):
...
             for build in [app] + get_saved_apps(app):
                 for form_unique_id in form_unique_ids:
-                    set_xmlns_on_form(
-                        form_unique_id,
-                        unique_id_to_xmlns_map[form_unique_id],
-                        build,
-                        log_file,
-                        dry_run
-                    )
+                    if unique_id_to_xmlns_map.get(form_unique_id):
```
Comment (millerdev) [inline corehq/apps/cleanup/management/commands/fix_forms_with_missing_xmlns.py]: the idiomatic way to do this is `form_unique_id in unique_id_to_xmlns_map`, which evaluates to true even when the value for that key evaluates to false

Comment (benrudolph) [inline corehq/apps/cleanup/management/commands/fix_forms_with_missing_xmlns.py]: ah right don't know why i did that. thank you. for this case it doesn't matter (and it's a one off), but will keep it in mind

**Comment by NoahCarnahan** [comment]
Hmm, so I think this doesn't actually have to do with forms being deleted. These two dictionaries don't have the best names, but this is what they represent:
Each entry in `unique_id_to_xmlns_map` means "We will now be using the new xmlns of <value> for form <key>".
Each entry in `app_to_unique_ids_map` means "App <key> has forms <value> that need their xmlns replaced in each build"
So it looks like the error you're seeing is basically that we've identified a form that needs fixing, but we've lost track of what xmlns we should be setting on the form (and this could be a problem, because we might have already set some the xmlns on some of the submissions that correspond to this form).

One thing I'm curious about is: do the keys that are missing in the `unique_id_to_xmlns_map` come from the log file, or from queries during the dry run?
If they come from the log file, that suggests that we might have done something wrong when logging what changes we made.

Another thing we could do is try to identify submissions that match these forms by searching ES for xforms with a matching name and app_id. If those submissions have xmlns, we might be able to determine the value we expect to be in unique_id_to_xmlns_map.

I'm happy to jump on a call if you want to go to voice about this.

**Comment by esoergel** [comment]
If it's just two forms, it might also be easier to do those manually - log that they're missing, then look at the corresponding apps to see if you can find a form that looks like these ones.

**Comment by benrudolph** [comment]
yeah so i was confused because the two ids that were missing, i couldn't find at all on HQ. which is what led me to believe it was some sort of deletion thing. i'll take a look at their build ids too. agree with ethan though that by and large we should just run this command and fix ~everything and then handle those last two cases manually

**Comment by esoergel** [comment]
If you have IDs that don't show up in `raw_couch`, they could be deleted, but if they're related to forms, they could also be `form.unique_id`s.  The form definitions tied to applications (as opposed to form instances submitted by mobile users) have a unique ID which doesn't correspond to any object in the database.

That is, `db.get(form_id)` will fail in any db - the form can be accessed by `Application.get(app_id).get_form(form_id)`

Just a guess though, I'm not sure if this was a context where that's possible.

**Comment by benrudolph** [comment]
yeah you're totally right. going to look to do that

**Comment by esoergel** [comment]
Relatedly, it was bothering me that it's hard to tell if a doc was deleted or if it's just missing, so I updated `raw_couch`: https://github.com/dimagi/commcare-hq/pull/11426

**Pull request merged**